File: backend/services/menu_service.py
from repository.menu_repository import MenuRepository
from DTO.menu_dto import MenuDto

class MenuService:
    @staticmethod
    def get_menu(menu_id):
        menu_item = MenuRepository.get_menu(menu_id)
        menu_data = MenuDto.get_menu_data_from_model(menu_item)
        return menu_data

    @staticmethod
    def create_menu(request_data):
        restaurant_id, dish_name, price, description = MenuDto.get_menu_data_from_request(request_data)
        new_menu_item = MenuRepository.add_menu(restaurant_id=restaurant_id, dish_name=dish_name, price=price, description=description)
        menu_data = MenuDto.get_menu_data_from_model(new_menu_item)
        return menu_data

    @staticmethod
    def update_menu(request_data, menu_id):
        restaurant_id, dish_name, price, description = MenuDto.get_menu_data_from_request(request_data)
        menu_item = MenuRepository.update_menu(menu_id=menu_id, restaurant_id=restaurant_id, dish_name=dish_name, price=price, description=description)
        if menu_item:
            menu_data = MenuDto.get_menu_data_from_model(menu_item)
            return menu_data
        return None

    @staticmethod
    def delete_menu(menu_id):
        success = MenuRepository.delete_menu(menu_id)
        return success
    
    @staticmethod
    def get_menus_for_restaurant(restaurant_id):
        menu_items = MenuRepository.get_menus_for_restaurant(restaurant_id)
        menu_data_list = [MenuDto.get_menu_data_from_model(menu_item) for menu_item in menu_items]
        return menu_data_list
    
    

File: frontend/foodstore/src/app/services/menu.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
@Injectable({
  providedIn: 'root'
})
export class MenuService {

  private apiUrl = 'http://127.0.0.1:4000/api';
  constructor(
    private http: HttpClient ,
  ) { }

  addMenu(menuData: any , restaurantId : any): Observable<any> {
    
    const payload ={
      'dish_name' : menuData.dishName ,
      'price' : menuData.price ,
      'description' : menuData.description ,
      'restaurant_id' : restaurantId
    }
    const url = `${this.apiUrl}/menu`;
    return this.http.post(url, payload );
  }

  getMenu(menuId : any) : Observable<any> {
    const url = `${this.apiUrl}/menu/${menuId}`
    return this.http.get(url)

  }

  updateMenu(menu : any) : Observable<any>{

    const url = `${this.apiUrl}/menu/${menu.menu_id}`
    return this.http.put(url , menu)

  }

  deleteMenu(menuId : any) : Observable<any>{

    const url = `${this.apiUrl}/menu/${menuId}`
    return this.http.delete(url)

  }
}
